tests/test_symbolic_primitives.py

```diff
@@ -90,19 +83,34 @@ def start_env():
             },
         ],
     }
-
-    gm.USE_GPU_DYNAMICS = True
-    gm.ENABLE_TRANSITION_RULES = False
-
     env = og.Environment(configs=config)
 
     return env
 
 
+def retrieve_obj_cfg(obj):
+    return {
+        "name": obj.name,
+        "category": obj.category,
+        "model": obj.model,
+        "prim_type": obj.prim_type,
+        "position": obj.get_position_orientation()[0],
+        "scale": obj.scale,
+        "abilities": obj.abilities,
+        "visual_only": obj.visual_only,
+    }
+
+
+def pytest_generate_tests(metafunc):
+    if "robot_type" in metafunc.fixturenames:
+        metafunc.parametrize("robot_type", ["Fetch", "Tiago"], scope="session")
+
+
 @pytest.fixture(scope="module")
-def shared_env():
-    """Load the environment just once using module scope."""
-    return start_env()
+def shared_env(robot_type):
+    """Create the environment once per session for each robot type."""
+    env = start_env(robot_type=robot_type)
+    return env
 
 
 @pytest.fixture(scope="function")
```

Comment:
Essentially, to fix the issues with the tests needing cleanup, all you need is the shared_env.scene.reset() here to actually reset back to the original state of the objects. I think that should already be happening for the most part? If not, find out why.